HHS/Head-Start-TTADP
# [TTAHUB-1029] only certain urls and other bugs

## Description of change
- Improved URL validation with a regex (see ticket for specs)
- Reorder objective status choices (moved suspended before closed)
- On the activity report, don't derive UI state from the status dropdown (fixes: "AR goals and objectives tab - TTA resources disappears when "suspend" status is chosen")
- Display validation on next steps when the fields are blank and the "Add next steps" button is clicked

## How to test
- Enter some URLs and make sure that the conditions are satisfied
- Confirm the objective statuses are in the right order
- Confirm that you can set an objective to suspended without losing sections of the UI
- Go the next steps page and try and add a next step without entering any data, you will see the validation messages

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1029


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
